File: backend/app/schemas/analysis.py
from pydantic import BaseModel

class CategoryFeedback(BaseModel):
    category: str
    details: list[str]

class AnalysisResult(BaseModel):
    score: int
    summary: str
    strengths: list[CategoryFeedback]
    improvements: list[CategoryFeedback]
    ats_notes: list[str]
    next_steps: list[str]

File: frontend/src/types/analysis.ts
export type CategoryFeedback = {
  category: string;
  details: string[];
};

export type AnalysisResult = {
  score: number;
  summary: string;
  strengths: CategoryFeedback[];
  improvements: CategoryFeedback[];
  ats_notes: string[];
  next_steps: string[];
};
